handlers/ruleset.go

```diff
@@ -110,3 +144,125 @@ func ValidateRuleSet(store kv.Store, ruleSetKey string) (int, *schema.RuleSet, e
 	}
 	return ruleSetVersion, ruleSet, nil
 }
+
+// UpdateRules updates an existing RuleSet by appending a diffRuleSet to it.
+func UpdateRules(store kv.TxnStore,
+	ruleSet, diffRuleSet *schema.RuleSet,
+	ruleSetKey string, ruleSetVersion int,
+	namespacesKey string, namespacesVersion int,
+	propDelay time.Duration,
+) error {
+	// There is nothing to do if the diff is nil
+	if diffRuleSet == nil {
+		return nil
+	}
+
+	if ruleSet == nil {
+		ruleSet = diffRuleSet
+	}
+
+	err := appendToRuleSet(ruleSet, diffRuleSet)
+	if err != nil {
+		return err
+	}
+
+	nowNs := time.Now().UnixNano()
+	ruleSet.LastUpdatedAt = nowNs
+	ruleSet.CutoverTime = NewCutoverNs(nowNs, ruleSet.CutoverTime, propDelay)
+
+	// Perform a transaction and only update if the namespaces version
```

Comment:
Can we move this logic to a function instead? We seem to apply the same logic during namespace creation.